src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -180,7 +184,7 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
 
     const { system_id, component_id } = mavlink_message.header
 
-    if (system_id != 1 || component_id != 1) {
+    if (system_id !== this.currentSystemId || component_id !== 1) {
```

Comment:
As we have this check here already, and the `onMavlinkPackage` funciton is called always after this line, we can remove the `if (component_id !== 1) return` check inside the specific-vehicle classes (e.g.: `ardusub.ts`, `arduplane.ts`, etc) and make the patch even simpler.